### PR TITLE
Correct k-NN settings and add more

### LINKED ISSUE #7642: [DOC] knn.advanced.filtered_exact_search_threshold is mentioned as a cluster settings but it's not
[This page](https://opensearch.org/docs/2.15/search-plugins/knn/settings/) lists `knn.advanced.filtered_exact_search_threshold` as a cluster setting. But if you try to apply it, you will get an error:
```
persistent setting [knn.advanced.filtered_exact_search_threshold], not recognized
```
After a some trial and error I dug [into code](https://github.com/opensearch-project/k-NN/blob/5139b163bc4a36aef7c99ce53d5e8b14c43eb768/src/main/java/org/opensearch/knn/index/KNNSettings.java#L81) and found it's actually an index level settings. The table need to be updated into 2 separated parts, or at least prefix index ones with `index.`.

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Closes #7642 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by jmazanec15 — APPROVED** [review]
LGTM

File: _search-plugins/knn/settings.md [hunk 1]
```diff
@@ -12,17 +12,28 @@ The k-NN plugin adds several new cluster settings. To learn more about static an
 
 ## Cluster settings
 
+The following table lists all available cluster-level k-NN settings. For more information about cluster settings, see [Configuring OpenSearch]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index/#updating-cluster-settings-using-the-api) and [Updating cluster settings using the API]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index/#updating-cluster-settings-using-the-api).
+
+Setting | Static/Dynamic | Default | Description
+:--- | :--- | :--- | :---
+`knn.plugin.enabled`| Dynamic | `true` | Enables or disables the k-NN plugin.
+`knn.algo_param.index_thread_qty` | Dynamic | `1` | The number of threads used for native library index creation. Keeping this value low reduces the CPU impact of the k-NN plugin, but also reduces indexing performance.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:20]:
```suggestion
`knn.algo_param.index_thread_qty` | Dynamic | `1` | The number of threads used for native library index creation. Keeping this value low reduces the CPU impact of the k-NN plugin but also reduces indexing performance.
```

File: _search-plugins/knn/settings.md [hunk 2]
```diff
@@ -12,17 +12,28 @@ The k-NN plugin adds several new cluster settings. To learn more about static an
...
+`knn.cache.item.expiry.enabled` | Dynamic | `false` | Whether to remove native library indexes that have not been accessed for a certain duration from memory.
+`knn.cache.item.expiry.minutes` | Dynamic | `3h` | If enabled, the idle time before removing a native library index from memory.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:22]:
```suggestion
`knn.cache.item.expiry.minutes` | Dynamic | `3h` | If enabled, the amount of idle time before a native library index is removed from memory.
```

File: _search-plugins/knn/settings.md [hunk 3]
```diff
@@ -12,17 +12,28 @@ The k-NN plugin adds several new cluster settings. To learn more about static an
...
+`knn.circuit_breaker.unset.percentage` | Dynamic | `75` | The native memory usage threshold for the circuit breaker. Memory usage must be below this percentage of `knn.memory.circuit_breaker.limit` for `knn.circuit_breaker.triggered` to remain `false`.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:23]:
```suggestion
`knn.circuit_breaker.unset.percentage` | Dynamic | `75` | The native memory usage threshold for the circuit breaker. Memory usage must be lower than this percentage of `knn.memory.circuit_breaker.limit` in order for `knn.circuit_breaker.triggered` to remain `false`.
```

File: _search-plugins/knn/settings.md [hunk 4]
```diff
@@ -12,17 +12,28 @@ The k-NN plugin adds several new cluster settings. To learn more about static an
...
+`knn.circuit_breaker.triggered` | Dynamic | `false` | True when memory usage exceeds the `knn.circuit_breaker.unset.percentage` value.
+`knn.memory.circuit_breaker.limit` | Dynamic | `50%` | The native memory limit for native library indexes. At the default value, if a machine has 100 GB of memory and the JVM uses 32 GB, the k-NN plugin uses 50% of the remaining 68 GB (34 GB). If memory usage exceeds this value, k-NN removes the least recently used native library indexes.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:25]:
```suggestion
`knn.memory.circuit_breaker.limit` | Dynamic | `50%` | The native memory limit for native library indexes. At the default value, if a machine has 100 GB of memory and the JVM uses 32 GB, then the k-NN plugin uses 50% of the remaining 68 GB (34 GB). If memory usage exceeds this value, then the plugin removes the native library indexes used least recently.
```

File: _search-plugins/knn/settings.md [hunk 5]
```diff
@@ -12,17 +12,28 @@ The k-NN plugin adds several new cluster settings. To learn more about static an
...
+`knn.memory.circuit_breaker.enabled` | Dynamic | `true` | Whether to enable the k-NN memory circuit breaker.
+`knn.model.index.number_of_shards`| Dynamic | `1` | The number of shards to use for the model system index, the OpenSearch index that stores the models used for Approximate Nearest Neighbor (ANN) search.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:27]:
```suggestion
`knn.model.index.number_of_shards`| Dynamic | `1` | The number of shards to use for the model system index, which is the OpenSearch index that stores the models used for approximate nearest neighbor (ANN) search.
```

File: _search-plugins/knn/settings.md [hunk 6]
```diff
@@ -12,17 +12,28 @@ The k-NN plugin adds several new cluster settings. To learn more about static an
...
+`knn.model.index.number_of_replicas`| Dynamic | `1` | The number of replica shards to use for the model system index. Generally, in a multi-node cluster, this should be at least 1 to increase stability.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:28]:
```suggestion
`knn.model.index.number_of_replicas`| Dynamic | `1` | The number of replica shards to use for the model system index. Generally, in a multi-node cluster, this value should be at least 1 in order to increase stability.
```

File: _search-plugins/knn/settings.md [hunk 7]
```diff
@@ -12,17 +12,28 @@ The k-NN plugin adds several new cluster settings. To learn more about static an
...
+`knn.model.cache.size.limit` | Dynamic | `10%` |  Model cache limit cannot exceed 25% of the JVM heap.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:29]:
```suggestion
`knn.model.cache.size.limit` | Dynamic | `10%` |  The model cache limit cannot exceed 25% of the JVM heap.
```

File: _search-plugins/knn/settings.md [hunk 8]
```diff
@@ -12,17 +12,28 @@ The k-NN plugin adds several new cluster settings. To learn more about static an
...
-`knn.advanced.filtered_exact_search_threshold`| null | The threshold value for the filtered IDs that is used to switch to exact search during filtered ANN search. If the number of filtered IDs in a segment is less than this setting's value, exact search will be performed on the filtered IDs.  
-`knn.faiss.avx2.disabled` | False | A static setting that specifies whether to disable the SIMD-based `libopensearchknn_faiss_avx2.so` library and load the non-optimized `libopensearchknn_faiss.so` library for the Faiss engine on machines with x64 architecture. For more information, see [SIMD optimization for the Faiss engine]({{site.url}}{{site.baseurl}}/search-plugins/knn/knn-index/#simd-optimization-for-the-faiss-engine).
+:--- | :--- | :--- 
+`index.knn.advanced.filtered_exact_search_threshold`| `null` | The threshold value for the filtered IDs that is used to switch to exact search during filtered ANN search. If the number of filtered IDs in a segment is less than this setting's value, exact search will be performed on the filtered IDs.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:38]:
```suggestion
`index.knn.advanced.filtered_exact_search_threshold`| `null` | The filtered ID threshold value used to switch to exact search during filtered ANN search. If the number of filtered IDs in a segment is lower than this setting's value, then exact search will be performed on the filtered IDs. 
```

File: _search-plugins/knn/settings.md [hunk 9]
```diff
@@ -12,17 +12,28 @@ The k-NN plugin adds several new cluster settings. To learn more about static an
...
-`knn.faiss.avx2.disabled` | False | A static setting that specifies whether to disable the SIMD-based `libopensearchknn_faiss_avx2.so` library and load the non-optimized `libopensearchknn_faiss.so` library for the Faiss engine on machines with x64 architecture. For more information, see [SIMD optimization for the Faiss engine]({{site.url}}{{site.baseurl}}/search-plugins/knn/knn-index/#simd-optimization-for-the-faiss-engine).
+:--- | :--- | :--- 
+`index.knn.advanced.filtered_exact_search_threshold`| `null` | The threshold value for the filtered IDs that is used to switch to exact search during filtered ANN search. If the number of filtered IDs in a segment is less than this setting's value, exact search will be performed on the filtered IDs. 
+`index.knn.algo_param.ef_search` | `100` | `ef` (or `efSearch`) is the size of the dynamic list for the nearest neighbors used during the search. Higher values of `ef` lead to a more accurate but slower search. `ef` cannot be set lower than the number of queried nearest neighbors `k`. `ef` can take any value between `k` and the size of the dataset.
```
Comment (natebower) [inline _search-plugins/knn/settings.md:39]:
```suggestion
`index.knn.algo_param.ef_search` | `100` | `ef` (or `efSearch`) represents the size of the dynamic list for the nearest neighbors used during a search. Higher `ef` values lead to a more accurate but slower search. `ef` cannot be set to a value lower than the number of queried nearest neighbors, `k`. `ef` can take any value between `k` and the size of the dataset. 
```

Comment (natebower) [inline _search-plugins/knn/settings.md:39]: the "total" size of the dataset?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

**Review by natebower — APPROVED** [review]
LGTM

**Pull request merged**